pkg/controllers/provisioning/controller.go

```diff
@@ -150,19 +150,25 @@ func requirements(instanceTypes []cloudprovider.InstanceType) []v1.NodeSelectorR
 		v1alpha5.LabelCapacityType: sets.NewString(),
 	}
 	for _, instanceType := range instanceTypes {
-		for _, offering := range instanceType.Offerings() {
-			supported[v1.LabelTopologyZone].Insert(offering.Zone)
-			supported[v1alpha5.LabelCapacityType].Insert(offering.CapacityType)
+		// filter out bare metal instance type unless required by provisioner requirements
+		if instanceType.IsBareMetal() &&
```

Comment:
rather than making this specific for baremetal, why not just allow any instance type if explicitly added to InstanceType requirements?  

I think if we change the AWS cloudprovider `GetInstanceTypes` to key off the passed in constraints (which it currently ignores), we can return InstanceTypes for the specific constraints. If the instance-type requirements are empty in the constrains, then we'll use the logic we use now. But if explicit InstanceTypes are specified with `In` we'd just return those instance types.